features/map.go

```diff
@@ -159,6 +140,84 @@ func isStorageMap(mt ebpf.MapType) bool {
 	case ebpf.SkStorage, ebpf.InodeStorage, ebpf.TaskStorage:
 		return true
 	}
-
 	return false
 }
+
+// HaveMapFlag probes the running kernel for the availability of the specified map flag.
+//
+// See the package documentation for the meaning of the error return value.
+func HaveMapFlag(flag uint32) (err error) {
+	defer func() {
+		// This closure modifies a named return variable.
+		err = wrapProbeErrors(err)
+	}()
+
+	return haveMapFlag(flag)
+}
+
+func haveMapFlag(flag uint32) error {
+	mc.Lock()
+	defer mc.Unlock()
+	err, ok := mc.mapFlags[flag]
+	if ok {
+		return err
+	}
+
+	attr, err := createMapFlagTypeAttr(flag)
+	if err != nil {
+		return err
+	}
+
+	fd, err := sys.MapCreate(attr)
```

Comment:
In general `HaveMapFlag` needs to do two feature probes:
1. Is the map type we use to probe supported?
2. Is the flag on that map type supported?

We know that all supported kernels have Array and Hash, so it's OK to skip the first check here and only check for (2). Instead of calling `convertProbeError` I'd therefore just check that `err == nil` to determine whether the flag is supported.

Reply:
Agreed, maybe you would like be to add a comment explaining why we skip (1) for now, but it may be necessary in the future ? I changed the convertProbeError to just check for `EINVAL` and convert it to `ErrNotSupported` in the latest commit

Reply:
Adding that comment would be good, thanks!